requests_cache/backends/base.py

```diff
@@ -152,6 +143,35 @@ def has_url(self, url: str) -> bool:
         """Returns `True` if cache has `url`, `False` otherwise. Works only for GET request urls"""
         return self.has_key(url_to_key(url, self.ignored_parameters))  # noqa: W601
 
+    def keys(self) -> Iterator[str]:
+        """Get all cache keys for redirects and (valid) responses combined"""
+        yield from self.redirects.keys()
+        for key, _ in self._get_valid_responses():
+            yield key
+
+    def values(self) -> Iterator[CachedResponse]:
+        """Get all valid response objects from the cache"""
+        for _, response in self._get_valid_responses():
+            yield response
+
+    def _get_valid_responses(self, delete_invalid=False) -> Iterator[Tuple[str, CachedResponse]]:
+        """Get all responses from the cache, and skip (+ optionally delete) any invalid ones that
+        can't be deserialized"""
+        keys_to_delete = []
+
+        for key in self.responses.keys():
+            try:
+                yield key, self.responses[key]
+            except Exception as e:
+                logger.debug(f'Unable to deserialize response with key {key}: {e}')
+                keys_to_delete.append(key)
+
+        # Delay deletion until the end, to slightly improve responsiveness when used as a generator
```

Comment:
I would say massively improve.

I was running this code over 40mb cache and it was running that long that I already fired strace to see WTH is it doing.

I can't sau how many entries the cache contained, because the method deleted entries.

From personal feeling it ran the method for 5-10minutes on RPI4.

Reply:
Was this with `remove_expired_responses()`? Yeah, at the moment requests-cache really isn't optimized for manually clearing items in bulk like this. There are several ways this could be made _much_ faster. If that's something you expect to be doing frequently, it would be worth making an issue for it.

Reply:
I mostly noticed the sqlite was opened million* of times.

I did not call the method myself. I was using the _get_valid_requests method to iterate over responses.

But as it effectively made whole cache empty, I did not have test no more to investigate further and dropped investigating the problem from there :) for me it was one time problem, that does not require followup. 

* not the exact number :)

Reply:
PR #240 likely made this a bit faster.

This will get more attention if/when someone wants to work on alternative cache eviction methods, for example an LRU-style cache (#116). The main change needed for that is adding an indexed expiration column that works consistently across all backends, which would also make `remove_expired_responses()` much faster.